src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory.cs

```diff
@@ -506,7 +509,35 @@ private IDynamicObjectCreationExpression CreateBoundDynamicObjectCreationExpress
             ITypeSymbol type = boundDynamicObjectCreationExpression.Type;
             Optional<object> constantValue = ConvertToOptional(boundDynamicObjectCreationExpression.ConstantValue);
             bool isImplicit = boundDynamicObjectCreationExpression.WasCompilerGenerated;
-            return new LazyDynamicObjectCreationExpression(name, applicableSymbols, arguments, argumentNames, argumentRefKinds, initializer, _semanticModel, syntax, type, constantValue, isImplicit);
+            return new LazyDynamicObjectCreationExpression(applicableSymbols, arguments, argumentNames, argumentRefKinds, initializer, _semanticModel, syntax, type, constantValue, isImplicit);
+        }
+
+        private IDynamicInvocationExpression CreateBoundDynamicInvocationExpressionOperation(BoundDynamicInvocation boundDynamicInvocation)
+        {
+            Lazy<IOperation> expression = new Lazy<IOperation>(() => Create(boundDynamicInvocation.Expression));
+            ImmutableArray<ISymbol> applicableSymbols = StaticCast<ISymbol>.From(boundDynamicInvocation.ApplicableMethods);
+            Lazy<ImmutableArray<IOperation>> arguments = new Lazy<ImmutableArray<IOperation>>(() => boundDynamicInvocation.Arguments.SelectAsArray(n => Create(n)));
```

Comment:
Consider extracting a helper method for creating a `Lazy<ImmutableArray<IOperation>>` from `ImmutableArray<BoundExpression>` since there appear to be many instances of this.

Reply:
Good idea, but there are large number of instances of this and I would prefer it to be a separate cleanup change, especially given we want to write an operation generator. I have added this suggestion to the issue tracking implementation of the operation generator: https://github.com/dotnet/roslyn/issues/20204#issue-235774159